WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -17,19 +17,97 @@ private struct QueryRef<Query> {
 /// Subclasses need to implement queriesChanged() and look at activeQueries and
 /// their internal state to decide if they need to act.
 ///
+/// If your `State` type confrorms to the `Codable` protocol, the `QueryStore`
+/// will transparently persist/load State and purge in-memory State when there are no
+/// longer any active `Queries` associated with the `Store`.
+////
+/// You can also manually force writing out the state to disk by calling `persistState()`.
+///
+
 open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
+
+    /// Closure used to log errors. Default implementation calls `NSLog`, clients may find it useful
+    /// to override it to use a custom logging method.
+    var logError: (String) -> Void = { error in
+        NSLog(error)
+    }
+
+    private let initialState: State
+
     fileprivate var activeQueryReferences = [QueryRef<Query>]() {
         didSet {
+            if activeQueryReferences.isEmpty, let encodableState = state as? Encodable, let url = try? type(of: self).persistenceURL() {
+                // If we don't have any active queries, and the `state` conforms to `Encodable`, let's use this as our cue to persist the data
+                // to disk and get rid of the in-memory cache.
+                do {
+                    try encodableState.saveJSON(at: url)
+                    inMemoryState = nil
+                } catch {
+                    logError("[QueryStore Error] \(error)")
```

Comment:
I haven't tested this, but I believe `type(of: self)` will give you the subclass name instead of using `QueryStore` here.